ansible/configs/ocp4-cluster/pre_infra.yml

```diff
@@ -6,12 +6,21 @@
   - step001
   - pre_infrastructure
   tasks:
+    - debug:
+        msg: "Step 000 Pre Infrastructure"
     - name: Ensure variables are set
       assert:
         that: "{{ item.0 }}"
         msg: "{{ item.1 }}"
       loop:
         - - ocp4_pull_secret is defined
           - ocp4_pull_secret variable must be defined
-    - debug:
-        msg: "Step 000 Pre Infrastructure - Dummy action"
+    - name: if windows_password is not defined, generate one
+      when: windows_password is not defined
```

Comment:
Does this need a check for cloud provider? "when: azure"?

Reply:
We can, but it doesn't hurt to just generate it and then not use it.